### PR TITLE
D8/9 Existing case not updated when contact is filled via autocomplete widget

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Existing case not updated when contact is filled via autocomplete widget

Before
----------------------------------------
Existing case not updated when contact is filled via autocomplete. To replicate -

- Create a case on contact A.
- Create a webform with Existing Contact Field, First name and last name.
- Enable Case section and set "Update Existing Case" field.
- Remove the default value in the existing contact field element to let the user fill it while submitting the form.
- Submit the webform with Existing contact = Contact A.
- Notice that a NEW case is created on the contact instead of updating the existing case.

After
----------------------------------------
Fixed. The existing case on the contact is updated.

Comments
----------------------------------------
D7 PR - https://github.com/colemanw/webform_civicrm/pull/509

### CODE REVIEW

**Comment by jitendrapurohit** [comment]
>RuntimeException: The provided database connection in SIMPLETEST_DB contains CiviCRM tables, use a different database.

Why test db is already containing civi tables 🤔? The test runs fine on local.

**Comment by KarinG** [comment]
Hi @jitendrapurohit - that sometimes happens when there are a lot of tests running. I just restarted - `rerun jobs` -> https://github.com/colemanw/webform_civicrm/actions/runs/657684553

File: tests/src/FunctionalJavascript/CiviCrmTestBase.php [hunk 1]
```diff
@@ -45,6 +45,12 @@ protected function cleanupEnvironment() {
...
       }
     }
+
+    // Drop case views if present. These are removed in the civi version 5.37.alpha1 https://github.com/civicrm/civicrm-core/pull/19642
+    // so these stmts wont be needed after few days when our test setup is configured to run on civi > 5.37.alpha1.
+    $civicrm_test_conn->query('DROP VIEW IF EXISTS civicrm_view_case_activity_recent;')->execute();
+    $civicrm_test_conn->query('DROP VIEW IF EXISTS civicrm_view_case_activity_upcoming;')->execute();
+
     $civicrm_test_conn->query('SET FOREIGN_KEY_CHECKS = 1;')->execute();
   }
```
Comment (KarinG) [inline tests/src/FunctionalJavascript/CiviCrmTestBase.php:56]: Ah - that was it! Excellent 👍

File: tests/src/FunctionalJavascript/WebformCivicrmTestBase.php [hunk 2]
```diff
@@ -271,4 +272,21 @@ protected function createIndividual() {
...
+    if (in_array($componentName, $enabledComponents)) {
+      // component is already enabled
+      return;
+    }
+    $enabledComponents[] = $componentName;
+    $this->utils->wf_civicrm_api('Setting', 'create', [
+      'enable_components' => $enabledComponents,
+    ]);
+  }
+
```
Comment (KarinG) [inline tests/src/FunctionalJavascript/WebformCivicrmTestBase.php:291]: Another helper function 👍 I'll try remember to use it!

**Comment by KarinG** [comment]
First Case Test! ✅ 🎉

**Pull request merged**